_posts/2019-09-20-lambda-to-openfaas.md

```diff
@@ -0,0 +1,330 @@
+---
+title: "Migrate Your AWS Lambda Functions to OpenFaaS"
+description: Burton explains the steps for migrating an existing AWS Lambda function to OpenFaaS
+date: 2019-09-10
+image: /images/lambda-to-openfaas/birds.jpg
+categories:
+  - Lambda
+  - tutorial
+  - examples
+author_staff_member: burton
+dark_background: false
+---
+
+AWS introduced the world to the idea of serverless architecture back in 2014 with the announcement of Lambda. The idea was that you could upload a zip file containing your code logic, and AWS would manage the infrastructure behind the scenes, billing you only for the time the code was being executed.
+
+In his talk [Welcome to Serverless 2.0](https://www.youtube.com/watch?v=JvXm-oHi5Mg), Alex Ellis described the new standard for serverless functions: Docker/OCI images exposing a port for execution, deployed to a Kubernetes cluster. This makes your functions completely portable and removes many of the limitations of the previous generation of serverless functions.
+
+In this post, We'll walk through the steps taken to migrate existing AWS Lambda functions to OpenFaaS compatible functions in order to allow easier local testing, familiar development process, and the freedom to use any number of other cloud providers without any code changes. We won't be moving all of the infrastructure, only the functions. We'll keep the DynamoDB table as well as the IAM roles/permissions to access that data.
+
+Without further ado, let's see what it takes to upgrade your AWS Lambda functions to OpenFaaS and Serverless 2.0 using OpenFaaS!
+
+## Prerequisites
+
+- Existing AWS Lambda functions (optional)
+  - For the tutorial, we'll be using the Lambda function in this [Github repository](https://github.com/burtonr/lambda-openfaas-blog)
+- OpenFaaS 
+  - Check the documentation for [how to deploy to Kubernetes](https://docs.openfaas.com/deployment/kubernetes/)
+- OpenFaaS CLI
+  - Check the documentation for [installation instructions](https://docs.openfaas.com/cli/install/)
+- IDE, or other text editor
+  - I recommend [Visual Studio Code](https://code.visualstudio.com/)
+
+## Getting Started
+
+To begin, let's look at the "hello world" sample function that gets created when you get started with Lambda. When you follow the [Getting Started Guide](https://docs.aws.amazon.com/lambda/latest/dg/getting-started-create-function.html) in the AWS documentation, a basic sample function is generated for you in Node.js 10.x as shown below:
+
+```js
+exports.handler = async (event) => {
+    // TODO implement
+    const response = {
+        statusCode: 200,
+        body: JSON.stringify('Hello from Lambda!'),
+    };
+    return response;
+};
+```
+
+Similarly, when you create a new function with the OpenFaaS CLI and the `node10-express` template, sample code is provided as shown here:
```

Comment:
Replace "similarly" for something else. It can be hard for International speakers to say.

"In a similar way" or "Let's examine the example for OpenFaaS"
...

"See how they are so similar?"

Reply:
Thank you for pointing that out!